content/en/blog/_posts/2024-07-18-kubernetes-1.31-deprecations-and-removals.md

```diff
@@ -0,0 +1,148 @@
+
+* Generally available (GA) or stable API versions may be marked as deprecated but must not be removed within a major version of Kubernetes.
+
+* Beta or pre-release API versions must be supported for 3 releases after the deprecation.
+
+* Alpha or experimental API versions may be removed in any release without prior deprecation notice.
+
+Whether an API is removed because a feature graduated from beta to stable or because that API simply did not succeed, all removals comply with this deprecation policy. 
+Whenever an API is removed, migration options are communicated in the [documentation](/docs/reference/using-api/deprecation-guide/).
+
+## A note about SHA-1 signature support fully going away in go1.24
+
+In [go1.18](https://go.dev/doc/go1.18#sha1) (released in March 2022), the crypto/x509 library started to reject certificates that were signed with a SHA-1 hash function. 
+While SHA-1 is established to be unsafe and publicly trusted Certificate Authorities have not issued SHA-1 certificates since 2015, there might still be cases in the context of Kubernetes where user-provided certificates are signed using a SHA-1 hash function through private authorities with them being used for Aggregated API Servers or webhooks. 
+If you have relied on SHA-1 based certificates, you must explicitly opt back into its support by setting `GODEBUG=x509sha1=1` in your environment.
+
+Given Go's [compatibility policy for GODEBUGs](https://go.dev/blog/compat), the `x509sha1` GODEBUG and the support for SHA-1 certificates will [fully go away in go1.24](https://tip.golang.org/doc/go1.23) which will be released in the first half of 2025. 
+If you rely on SHA-1 certificates, please start moving off them.
+
+Please see [Kubernetes issue #125689](https://github.com/kubernetes/kubernetes/issues/125689) to get a better idea of timelines around the support for SHA-1 going away, when Kubernetes releases plans to adopt go1.24, and for more details on how to detect usage of SHA-1 certificates via metrics and audit logging. 
+
+## Deprecations and Removals in Kubernetes 1.31
+
+### Deprecate kustomize from kubectl [KEP 4706](https://github.com/kubernetes/enhancements/issues/4706)
+
+In the early days of Kubernetes, when not that many tools in the field of declarative configurations existed, [kustomize](https://github.com/kubernetes-sigs/kustomize) was introduced, and to make life easier for kubectl users, it was included in kubectl by default.
+
+Over the past several years, the ecosystem around Kubernetes has grown significantly, resulting in a wider spread of tooling, including declarative configuration. 
+With these new options, users are able to use [kubectl plugins](/docs/tasks/extend-kubectl/kubectl-plugins/) to match their use cases with the best tool capable of resolving the problem at hand.
+
+With Kubernetes v1.31, kustomize will be deprecated from kubectl. It will be removed in a future release. 
```

Comment:
I do not agree with exposing the risk of this being removed in a disruptive way to end users yet. There were objections on the enhancements issue from myself and others (including the kustomize subproject maintainer) that questioned the reasons for the removal. Until it is actually settled that it will be removed, and there is actionable guidance for current users, I do not think we should surface this.

Reply:
/hold
on telling users kustomize will be removed

Reply:
@liggitt thank you for the feedback. We've removed the section about KEP 4706 based on this.

If anything changes and there is more consensus on this deprecation, please let us know and we can include it in the main release blog. 

/hold cancel

cc: @neoaggelos

Reply:
@a-mccarthy Just a heads up: Earlier today, [KEP-4706](https://github.com/kubernetes/enhancements/issues/4706) was removed from the current v1.31 release cycle and will be targeted for a future release after further discussions ([reference](https://github.com/kubernetes/enhancements/issues/4706#issuecomment-2233048060)).  It looks like we're all set for now!